openmdao/solvers/linear/petsc_ksp.py

```diff
@@ -214,6 +214,9 @@ def _declare_options(self):
                              desc='Number of iterations between restarts. Larger values increase '
                              'iteration cost, but may be necessary for convergence')
 
+        self.options.declare('precon_side', default='right', values=['left', 'right'],
```

Comment:
Any reason to prefer right preconditioning over left? If I recall correctly, more KSP solvers support only left than support only right.